include/glow/Runtime/StatsExporter.h

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_RUNTIME_STATSEXPORTER_H
+#define GLOW_RUNTIME_STATSEXPORTER_H
+
+#include "llvm/ADT/StringRef.h"
+
+namespace glow {
+
+/// Interface for exporting runtime statistics.  The base implementation
+/// delegates to any subclass registered via `registerStatsExporter`.
+class StatsExporter {
+public:
+  /// Register a stats exporter.
+  static void registerStatsExporter(StatsExporter *exporter);
+
+  /// Dtor.
+  virtual ~StatsExporter() {}
+
+  /// Add value to a quantile stat.
+  virtual void addQuantileValue(llvm::StringRef key, double value);
+
+  /// Set a counter.
+  virtual void incrementCounter(llvm::StringRef key, int64_t value);
```

Comment:
Maybe make a note that these could be called concurrently

Reply:
What's our expectation on threading here? Do we need to use atomics/mutexes?

Reply:
Right, these can absolutely be called concurrently.  I'd expect whatever loggers are registered to handle concurrency properly (in FB's case, fb303 does).